### PR TITLE
refactor: consolidate allow/exclude interval

### PR DESCRIPTION
**Pull request by s22chan**

### CODE REVIEW

**Comment by ovesh** [comment]
Some classes inherit from sam_line_parser (e.g read_alignments::builder), can we get rid of that now?

File: src/interval.cpp [hunk 1]
```diff
@@ -20,4 +27,34 @@ string interval_t::as_str() const
...
 pos_t as_pos(std::string_view s) { return as_int(s); }
 
+interval_filter::interval_filter(interval_filter::interval_fn_t validate_interval)
+: _validate_interval(validate_interval)
+{
+}
+
+void interval_filter::allow(interval_t i)
+{
+	if (_validate_interval)
```
[the remark this answers is not in the document]
Comment (s22chan) [reply src/interval.cpp:37]: {} implies it doesn't have a value

**Comment by s22chan** [comment]
> Some classes inherit from sam_line_parser (e.g read_alignments::builder), can we get rid of that now?

that's the class that does the actual parsing of the SAM. What would you suggest?

**Comment by ovesh** [comment]
> > Some classes inherit from sam_line_parser (e.g read_alignments::builder), can we get rid of that now?
> 
> that's the class that does the actual parsing of the SAM. What would you suggest?

Oh right, I didn't notice these builders are literally parsing sam files.

**Review by ovesh — APPROVED** [review]

**Pull request merged**